src/components/core/message-listener.ts

```diff
@@ -0,0 +1,96 @@
+    'Illegal mix of collations':
+        "you're database has mixed up collations, to fix this go into your database and at the `Collation` section change it to the collation that the players database uses. \nRead More at <https://docs.qbox.re/faq#error-sql-cant-create-table-xyz-foreign-key-constraint-is-incorrectly-formed>",
+    'rate limit exceeded':
+        "Don't worry, this is not an issue with our framework, instead a limitation on github, for its api to be able to be free. Please Wait 3-5 minutes or Try changing your IP address. \nRead More at <https://docs.qbox.re/faq#txadmin-installation-task-failed-response-code-403-rate-limit-exceeded>",
+};
+
+async function parseImage(url: string) {
+    const response = await fetch(apiUrl, {
+        method: 'POST',
+        headers: {
+            apikey: apiKey,
+            'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        body: new URLSearchParams({
+            language: 'eng',
+            isOverlayRequired: 'true',
+            url: url,
+            iscreatesearchablepdf: 'false',
+            issearchablepdfhidetextlayer: 'false',
+        }),
+    });
+    const data: any = await response.json();
+    if (data.ParsedResults && data.ParsedResults.length > 0) {
+        const parsedResult = data.ParsedResults[0];
+        let parsedText = parsedResult.ParsedText || null;
+        return parsedText;
+    } else {
+        return null;
+    }
+}
+
+async function parseMessage(url: string) {
+    let parsedText = await parseImage(url);
+    if (!parsedText) {
+        return undefined;
+    }
+    for (const key in errorMessages) {
+        if (parsedText.toLowerCase().includes(key.toLowerCase())) {
+            return errorMessages[key as keyof typeof errorMessages];
+        }
+    }
+    return undefined;
+}
+
+export const messageListener = {
+    name: GatewayDispatchEvents.MessageCreate,
+    type: 'on',
+    async execute({ api, data: message }) {
+        if (message.attachments.length > 0) {
+            message.attachments.forEach(async ({ url }) => {
+                const isImage =
+                    url.includes('.png') ||
+                    url.includes('.jpg') ||
+                    url.includes('.jpeg');
+                console.log(isImage);
+                if (isImage) {
+                    await api.channels.addMessageReaction(
+                        message.channel_id,
+                        message.id,
+                        '🔍',
+                    );
+                    let result = await parseMessage(url);
+                    await api.channels.deleteOwnMessageReaction(
+                        message.channel_id,
+                        message.id,
+                        '🔍',
+                    );
+
+                    if (!result) {
+                        return;
+                    } else {
```

Comment:
Let's remove this else statement as it's unnecessary.

Reply:
okayy!